src/nunit.analyzers/DiagnosticSuppressors/NonNullableFieldIsUninitializedSuppressor.cs

```diff
@@ -0,0 +1,81 @@
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using NUnit.Analyzers.Constants;
+
+namespace NUnit.Analyzers.DiagnosticSuppressors
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class NonNullableFieldIsUninitializedSuppressor : DiagnosticSuppressor
+    {
+        internal static readonly SuppressionDescriptor NullableFieldInitializedInSetUp = new SuppressionDescriptor(
+            id: AnalyzerIdentifiers.NonNullableFieldIsUninitialized,
+            suppressedDiagnosticId: "CS8618",
+            justification: "Field is initialized in SetUp or OneTimeSetUp method");
+
+        public override ImmutableArray<SuppressionDescriptor> SupportedSuppressions { get; } =
+            ImmutableArray.Create(NullableFieldInitializedInSetUp);
+
+        public override void ReportSuppressions(SuppressionAnalysisContext context)
+        {
+            foreach (var diagnostic in context.ReportedDiagnostics)
+            {
+                SyntaxNode? node = diagnostic.Location.SourceTree?.GetRoot(context.CancellationToken)
+                                                                  .FindNode(diagnostic.Location.SourceSpan);
+
+                if (node is null)
+                {
+                    continue;
+                }
+
+                string fieldName = node.ToString();
+
+                var classDeclaration = node.Ancestors().OfType<ClassDeclarationSyntax>().First();
+                var methods = classDeclaration.Members.OfType<MethodDeclarationSyntax>().ToArray();
+
+                foreach (var method in methods)
+                {
+                    var allAttributes = method.AttributeLists.SelectMany(list => list.Attributes.Select(a => a.Name.ToString()))
+                                                             .ToImmutableHashSet();
+                    if (allAttributes.Contains("SetUp") || allAttributes.Contains("OneTimeSetUp"))
+                    {
+                        // Find (OneTime)SetUps method and check for assignment to this field.
+                        if (FieldIsAssignedIn(method, fieldName))
+                        {
+                            context.ReportSuppression(Suppression.Create(NullableFieldInitializedInSetUp, diagnostic));
+                        }
+                    }
+                }
+            }
+        }
+
+        private static bool FieldIsAssignedIn(MethodDeclarationSyntax method, string fieldName)
+        {
+            if (method.Body is null)
+            {
+                return false;
+            }
+
+            foreach (var statement in method.Body.Statements)
+            {
+                if (statement is ExpressionStatementSyntax expressionStatement &&
+                    expressionStatement.Expression is AssignmentExpressionSyntax assignmentExpression)
+                {
+                    if (assignmentExpression.Left.ToString() == fieldName)
```

Comment:
Perhaps we should also handle when the left-side is prefixed with `this`, e.g. `this.someVariable`?

Reply:
I added two test cases, but the one advantage of text based comparison is that the code already handles `this.member` as the left part of the assignment.
The downside it, it doesn't handle inconsistent use of the `this` prefix, e.g. using it in the assignment, but not in references. But that would already be flagged by the compiler (IDE0009).